akka-remote/src/main/scala/akka/remote/artery/ArteryTcpTransport.scala

```diff
@@ -0,0 +1,352 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+
+import akka.Done
+import akka.NotUsed
+import akka.actor.ExtendedActorSystem
+import akka.dispatch.ExecutionContexts
+import akka.event.Logging
+import akka.remote.RemoteActorRefProvider
+import akka.remote.RemoteTransportException
+import akka.remote.artery.Decoder.InboundCompressionAccess
+import akka.remote.artery.compress._
+import akka.stream.Attributes
+import akka.stream.Attributes.LogLevels
+import akka.stream.FlowShape
+import akka.stream.Graph
+import akka.stream.KillSwitches
+import akka.stream.SharedKillSwitch
+import akka.stream.SinkShape
+import akka.stream.scaladsl.Broadcast
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Framing
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.MergeHub
+import akka.stream.scaladsl.Partition
+import akka.stream.scaladsl.RestartFlow
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.Tcp
+import akka.stream.scaladsl.Tcp.ServerBinding
+import akka.util.ByteString
+import akka.util.OptionVal
+
+/**
+ * INTERNAL API
+ */
+private[remote] class ArteryTcpTransport(_system: ExtendedActorSystem, _provider: RemoteActorRefProvider,
+                                         tlsEnabled: Boolean)
+  extends ArteryTransport(_system, _provider) {
+  import ArteryTransport._
+  import FlightRecorderEvents._
+
+  @volatile private var inboundKillSwitch: SharedKillSwitch = KillSwitches.shared("inboundKillSwitch")
+  private var serverBinding: Option[Future[ServerBinding]] = None
+
+  private val sslEngineProvider: OptionVal[SSLEngineProvider] =
+    // FIXME load from config
+    if (tlsEnabled) OptionVal.Some(new ConfigSSLEngineProvider(system))
+    else OptionVal.None
+
+  override protected def startTransport(): Unit = {
+    // FIXME new TCP events
+    topLevelFREvents.loFreq(Transport_AeronStarted, NoMetaData)
```

Comment:
Could go in the superclass after calling startTransport

Reply:
I'll fix all FRE events today, I agree that same could be used here